internal/cmd/local/command_kafka_start.go

```diff
@@ -158,20 +154,25 @@ func (c *Command) prepareAndSaveLocalPorts(cmd *cobra.Command, isTest bool) erro
 			ControllerPort: "2999",
 		}
 	} else {
-		freePorts, err := freeport.GetFreePorts(4)
+		freePorts, err := freeport.GetFreePorts(3)
 		if err != nil {
 			return err
 		}
 
 		c.Config.LocalPorts = &v1.LocalPorts{
-			KafkaRestPort:  strconv.Itoa(freePorts[0]),
-			PlaintextPort:  strconv.Itoa(freePorts[1]),
-			BrokerPort:     strconv.Itoa(freePorts[2]),
-			ControllerPort: strconv.Itoa(freePorts[3]),
+			PlaintextPort:  strconv.Itoa(freePorts[0]),
+			BrokerPort:     strconv.Itoa(freePorts[1]),
+			ControllerPort: strconv.Itoa(freePorts[2]),
 		}
 
 		if kafkaRestPort, err := cmd.Flags().GetString("kafka-rest-port"); err == nil && kafkaRestPort != "" {
 			c.Config.LocalPorts.KafkaRestPort = kafkaRestPort
+		} else {
+			freePort, err := freeport.GetFreePort()
+			if err != nil {
+				return err
+			}
+			c.Config.LocalPorts.KafkaRestPort = strconv.Itoa(freePort)
```

Comment:
Let's print a warn-level message here that says something like: "default kafka rest port 8082 is in use, using XXXX instead"?

Reply:
👍

Reply:
Not sure if you took this suggestion, since it looks like we don't tell the user that port "8082" is taken.

I think the majority of the time, if 8082 is already in use, it probably means that the user is already running an instance of Kafka REST, and they probably want to shut it down to avoid confusion.

Reply:
i added an extra step of validating that port and (potentially) customized plaintext port.